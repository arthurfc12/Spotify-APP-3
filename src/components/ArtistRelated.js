import React, { Component, useEffect, useState } from 'react';
import axios from 'axios';
import { Link } from 'react-router-dom';
import { Button } from '@material-ui/core';

export default function ArtistRelated(props) {
  const requestOptions = {
    headers: {
      Authorization: 'Bearer ' + props.token,
    },
  };
  const [topArtists, setTopArtist] = useState('');
  const [artistSelected, setArtistSelected] = useState('');
  const [ArtistRelatedList, setArtistRelatedList] = useState([]);
  const [vizualization, setVizualization] = useState(false);

  useEffect(() => {
    axios
      .get('https://api.spotify.com/v1/me/top/artists', requestOptions)
      .then((response) => {
        setTopArtist(response.data);
      });
  }, []);

  function getArtistList(artistId) {
    console.log('artistId:', artistId);
    axios
      .get(
        `https://api.spotify.com/v1/artists/${artistId}/related-artists`,
        requestOptions
      )
      .then((response) => {
        console.log(
          'related-artists:',
          JSON.stringify(response.data.artists.slice(0, 2))
        );
        setArtistRelatedList(response.data.artists.slice(0, 2));
      });
  }

  return !vizualization ? (
    <div style={{ display: 'flex', color: '#ffffff', flexDirection: 'column' }}>
      <p style={{ textAlign: 'left', fontSize: '1.6rem', fontWeight: '900' }}>
        Artistas Relacionados
      </p>
      <p style={{ textAlign: 'center', fontSize: '1.3rem' }}>
        Selecione um artista para ver os artistas relacionados:
      </p>
      {topArtists.items !== undefined &&
        Object.values(topArtists.items)?.map((item, key) => {
          return (
            <>
              <button
                className='gameOpt'
                key={`${key}-${item.id}`}
                onClick={() => {
                  console.log('oi');
                  getArtistList(item.id);
                  setArtistSelected(item.name);
                  setVizualization(true);
                }}
                style={{
                  height: 'fit-content',
                  backgroundColor: 'transparent',
                  border: 'none',
                  color: '#ffffff',
                  cursor: 'pointer',
                }}
              >
                <div
                  style={{
                    backgroundColor: '#222222',
                    height: '10rem',
                    width: '90%',
                    boxShadow: '0 0 10px #000000',
                    margin: '0.5rem',
                    display: 'flex',
                    justifyContent: 'space-between',
                    alignItems: 'center',
                    padding: '1rem',
                    borderRadius: '1rem',
                  }}
                >
                  <div>
                    <p style={{ fontSize: '1.5rem' }}>{item.name}</p>
                    <p style={{ fontSize: '1.2rem' }}>
                      {item.followers.total} followers
                    </p>
                  </div>
                  <img
                    style={{ height: '8rem', borderRadius: '50%' }}
                    src={Object.values(item.images)[2].url}
                  />
                </div>
              </button>
            </>
          );
        })}
    </div>
  ) : (
    <div style={{ display: 'flex', color: '#ffffff', flexDirection: 'column' }}>
      <Button
        className='gameOpt'
        style={{
          width: 'fit-content',
          padding: '0.1rem 2rem',
          fontSize: '1rem',
          backgroundColor: 'transparent',
          color: '#ffffff',
          fontWeight: '900',
          borderRadius: '10rem',
          margin: '2rem 0rem',
        }}
        onClick={() => {
          setVizualization(false);
        }}
      >
        Voltar
      </Button>
      <p style={{ textAlign: 'left', fontSize: '1.6rem', fontWeight: '900' }}>
        Artistas relacionados a {artistSelected} :
      </p>
      {ArtistRelatedList !== undefined &&
        ArtistRelatedList?.map((item, key) => {
          return (
            <>
              <div
                key={`${key}-${item.id}`}
                style={{
                  backgroundColor: '#222222',
                  height: '10rem',
                  width: '90%',

                  margin: '0.5rem',
                  display: 'flex',
                  justifyContent: 'space-between',
                  alignItems: 'center',
                  padding: '1rem',
                  borderRadius: '1rem',
                  color: '#ffffff',
                }}
              >
                <div>
                  <p style={{ fontSize: '1.5rem' }}>{item.name}</p>
                  <p style={{ fontSize: '1.2rem' }}>
                    {item.followers.total} followers
                  </p>
                </div>
                <img
                  style={{ height: '8rem', width: '8rem', borderRadius: '50%' }}
                  src={Object.values(item.images)[2].url}
                />
              </div>
            </>
          );
        })}
    </div>
  );
}
