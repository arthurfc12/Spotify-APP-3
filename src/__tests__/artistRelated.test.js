import React from 'react';
import { render, wait, cleanup, waitFor } from '@testing-library/react';
import ArtistRelated from '../components/ArtistRelated';
import { configure, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import mockedAxios from 'axios';

configure({ adapter: new Adapter() });

afterEach(cleanup);

it('should show artist related', async () => {
  const data = {
    artistSelected: '7185Q95lPFld0aoPqO6e0U',
    ArtistRelatedList: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/50nVKm0xa3p8KYXXssG2ym',
        },
        followers: { href: null, total: 7744 },
        genres: ['chillhop', 'lo-fi beats'],
        href: 'https://api.spotify.com/v1/artists/50nVKm0xa3p8KYXXssG2ym',
        id: '50nVKm0xa3p8KYXXssG2ym',
        images: [
          {
            height: 640,
            url:
              'https://i.scdn.co/image/a9dd952cbaa9bddf0fe03432b10be8780de2838d',
            width: 640,
          },
          {
            height: 320,
            url:
              'https://i.scdn.co/image/3494cb26cf678a71433484c9dff86618065fcac1',
            width: 320,
          },
          {
            height: 160,
            url:
              'https://i.scdn.co/image/d1ae2cac2332450a1c77a822390b40de8f38a715',
            width: 160,
          },
        ],
        name: 'Monma',
        popularity: 63,
        type: 'artist',
        uri: 'spotify:artist:50nVKm0xa3p8KYXXssG2ym',
      },
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/4C7NcNb9V6lakzMGHQlm8i',
        },
        followers: { href: null, total: 5299 },
        genres: ['lo-fi beats'],
        href: 'https://api.spotify.com/v1/artists/4C7NcNb9V6lakzMGHQlm8i',
        id: '4C7NcNb9V6lakzMGHQlm8i',
        images: [
          {
            height: 640,
            url:
              'https://i.scdn.co/image/b0aaba50c1bba08b36a7d0206d7e98628a5619a3',
            width: 640,
          },
          {
            height: 320,
            url:
              'https://i.scdn.co/image/0b29093fa3a5e26f8e3a6b2be23f65e50b230cf7',
            width: 320,
          },
          {
            height: 160,
            url:
              'https://i.scdn.co/image/31088f40bfd925cd87998db8b02b2d2208248373',
            width: 160,
          },
        ],
        name: 'Tom Doolie',
        popularity: 65,
        type: 'artist',
        uri: 'spotify:artist:4C7NcNb9V6lakzMGHQlm8i',
      },
    ],
  };

  mockedAxios.get.mockResolvedValueOnce(data);
  const app = shallow(<ArtistRelated />);

  // expect(mockedAxios.get).toHaveBeenCalledTimes(1);

  // expect(['7185Q95lPFld0aoPqO6e0U']).toContain(app.find('id').at(0).text());
  // await waitFor(() => {
  // });

  expect(toJson(app)).toMatchSnapshot();
});
