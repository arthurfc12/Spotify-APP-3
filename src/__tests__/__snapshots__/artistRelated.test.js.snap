// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show artist related 1`] = `
<div
  style={
    Object {
      "color": "#ffffff",
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <p
    style={
      Object {
        "fontSize": "1.6rem",
        "fontWeight": "900",
        "textAlign": "left",
      }
    }
  >
    Artistas Relacionados
  </p>
  <p
    style={
      Object {
        "fontSize": "1.3rem",
        "textAlign": "center",
      }
    }
  >
    Selecione um artista para ver os artistas relacionados:
  </p>
</div>
`;
